docs/dev/Lua API.rst

```diff
@@ -3928,6 +3928,211 @@ Examples:
     end
     unit.body.blood_count = math.min(unit.body.blood_max, unit.body.blood_count + healAmount)
 
+DateTime
+========
+
+A module to display, compare, and manipulate dates and times for the dwarven calendar.
+The *DateTime* and *Duration* objects in this module support *addition*, *subtraction*,
+and *<, <=, >, >=, ==* equality operators
+
+DateTime class
+==============
+
+* ``DateTime{ year = ..., year_tick = ..., ticks_per_day = ...}``
+
+  A DateTime object is constructed with the following properties:
+
+  :year: A year. Defaults to zero.
+  :year_tick: A year tick. Defaults to zero.
+  :ticks_per_day: The rate at which time passes in a day.
```

Comment:
I'm not thrilled with the ticks_per_day name. Originally ticks_per_day was going to be a mode (I believe (df.game_mode is an enum with DWARF=0, ADVENTURE=1, AND LEGENDS=3). I think you mentioned letting the tick rate be customizable (if can change rate of time in game, then tick rate in a DateTime object may need to coincide).

We can substitute ticks_per_day for mode in the docs, but keep ticks_per_day (unexposed) as an object property so that setTickRate() can change it?